### PR TITLE
Getting rid of memoization on many properties and methods

### PR DESCRIPTION
**Pull request by iskandr**
Partially addresses: https://github.com/hammerlab/pyensembl/issues/150

We're still leaking an unbounded amount of memory to memoization but now it's at least being centralized through a few "important" methods such as those which get Gene, Transcript, and Exon objects by ID or the database query helpers.

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/pyensembl/151)

### CODE REVIEW

File: pyensembl/gtf.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def __init__(self, gtf_path, cache_directory_path=None):
 
     def __eq__(self, other):
         return (
-            isinstance(other, GTF) and
+            (other.__class__ == GTF) and
```
Comment (timodonnell) [inline pyensembl/gtf.py]: not important, but just noting I think this change means you can't subclass GTF and have equality work. Not sure if that's intentional

File: pyensembl/sequence_data.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def __contains__(self, sequence_id):
 
     def __eq__(self, other):
         return (
-            isinstance(other, SequenceData) and
+            (other.__class__ is SequenceData) and
```
Comment (timodonnell) [inline pyensembl/sequence_data.py]: ditto to above (I think `is` is better than `==` tho)

**Comment by timodonnell** [comment]
LGTM

File: pyensembl/gtf.py (hunk 1 above)
Comment (iskandr) [inline pyensembl/gtf.py]: Hard to imagine when we'd want a sub-class of GTF to equal to a GTF instance, seemed safer to restrict equality to exactly the same class. .

**Pull request merged**